### PR TITLE
2774 spike gradients driven by variables

### LINKED ISSUE #2774: [Spike]: Gradients driven by variables
Let's get variable support in for gradients. Figma now added support for this

https://www.figma.com/plugin-docs/updates/2024/04/24/version-1-update-92/

Might be a good one for the quality of life track.

### PR DESCRIPTION
**Pull request by robinhoodie0823**
### Why does this PR exist?
Current plugin can not assign color variables to gradient's color stops due to previous figma plugin does not support it.

Closes #2774 

### What does this pull request do?
Updated figma plugin has `boundVariables` field with `color` field to support color variable assigning to gradient's color stops.
Updated `gradientStops` with color variables Ids in it.

### Additional Notes (if any)

https://github.com/tokens-studio/figma-plugin/assets/103296157/35b8d03e-482b-48aa-981d-90dc40667e25

### CODE REVIEW

File: packages/tokens-studio-for-figma/package.json [hunk 1]
```diff
@@ -178,7 +177,6 @@
     "@types/uuid": "^9.0.2",
     "@typescript-eslint/eslint-plugin": "^6.11.0",
     "@typescript-eslint/parser": "^6.11.0",
-    "0x": "^5.6.0",
```
Comment (macintoshhelper) [inline packages/tokens-studio-for-figma/package.json:181]: `0x` seems to be removed? It is used in a benchmark script

Comment (robinhoodie0823) [reply packages/tokens-studio-for-figma/package.json:181]: Added

File: packages/tokens-studio-for-figma/src/plugin/setColorValuesOnTarget.ts [hunk 2]
```diff
@@ -5,6 +5,16 @@ import { defaultTokenValueRetriever } from './TokenValueRetriever';
 import { ColorPaintType, tryApplyColorVariableId } from '@/utils/tryApplyColorVariableId';
 import { unbindVariableFromTarget } from './unbindVariableFromTarget';
 
+const getReferenceTokensFromGradient = (rawValue: string): string[] => {
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/plugin/setColorValuesOnTarget.ts:8]: can you extract this into its own file and write a test? this should be able to properly extract values like

linear-gradient(90deg, {accent.bg} 0%, {accent.default} 100%) = [accent.bg, accent.default]
linear-gradient(90deg, {accent.bg} 0%, {accent.default} 50%, #ff0000 100%) = [accent.bg, accent.default]
linear-gradient({accent.bg} 0%, {accent.default} 100%) = [accent.bg, accent.default]

Comment (six7) [reply packages/tokens-studio-for-figma/src/plugin/setColorValuesOnTarget.ts:8]: because this fails today:

image

**Review by six7 — CHANGES_REQUESTED** [review]

**Review by six7 — APPROVED** [review]
🎉

**Pull request merged**